modules/apps/frontend-taglib/frontend-taglib-soy/src/main/java/com/liferay/frontend/taglib/soy/servlet/taglib/TemplateRendererTag.java

```diff
@@ -145,7 +145,7 @@ public void setComponentId(String componentId) {
 	}
 
 	public void setContext(Map<String, Object> context) {
-		_context = context;
+		_context = SoyContextFactoryUtil.createSoyContext(context);
```

Comment:
We might be getting passed a `SoyContext` already (some apps are doing it)... should we check that here to avoid the extra initialization, or it's fine to always go through this again?